### PR TITLE
feat: upgrade to node 18

### PR DESCRIPTION
**Pull request by Yakitrak**
## Describe your changes

Ticket: https://pins-ds.atlassian.net/browse/ASB-1796

This PR upgrades node from 16 to 18. All unit tests and e2e tests are passing which gives us some confidence nothing has broken but please check anything that you might suspect could be broken from the file changes. The changes include

- [x]  Update nvmrc
- [x]  Update package.jsons
- [x]  Update Dockerfiles + Docker compose node images
- [x]  Update azure files 
- [x]  Run `npm update`
- [x]  Run `npm audit fix`
- [x]  Update documentation
- [x] Fix broken tests

To use node 18 locally, follow the steps on the documentation. Tldr:
```
nvm install 18.18.0
nvm use 18.18.0
nvm alias default 18
```
For the first time. you will also need to remove node_modules (from root and all the packages) and then do `npm install` from the root.

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: packages/back-office-subscribers/package.json [hunk 1]
```diff
@@ -9,7 +9,10 @@
 	},
 	"dependencies": {
 		"axios": "^1.4.0",
-		"lodash.pick": "^4.4.0"
+		"express": "4.18.2",
+		"lodash.pick": "^4.4.0",
+		"multi-semantic-release": "^2.13.0",
+		"sequelize": "6.33.0"
```
Comment (JamesP-KC) [inline packages/back-office-subscribers/package.json:15]: don't think some of these dependencies are needed in `back-office-subscribers` - definitely not using `express`, or `sequelize` in this package. I presume `multi-semantic-release` should be in the root `package.json` too? might even be a devDependency?

File: packages/applications-service-api/src/controllers/subscriptions.js [hunk 2]
```diff
@@ -78,6 +78,7 @@ const getApplicationOrThrow = async (caseReference) => {
 
 const validateSubscriptionDate = (subscriptionCreatedAt) => {
 	const expiryTime = moment(subscriptionCreatedAt).add(48, 'hours');
+	console.log({ expiryTime, now: moment() });
```
Comment (JamesP-KC) [inline packages/applications-service-api/src/controllers/subscriptions.js:81]: is this still needed?

File: package.json [hunk 3]
```diff
@@ -85,6 +85,9 @@
 		]
 	},
 	"dependencies": {
-		"@prisma/client": "^4.12.0"
+		"@prisma/client": "^4.12.0",
+		"azurite": "^3.26.0",
```
Comment (JamesP-KC) [inline package.json:89]: i think npm has done something funky with the dependencies, as i don't think `azurite` should be a non-dev dependency. I think it's only really needed as a devDependency in `back-office-subscribers`

File: packages/forms-web-app/package.json [hunk 4]
```diff
@@ -57,6 +57,7 @@
 		"pino": "^6.13.2",
 		"proj4": "^2.9.0",
 		"redis": "^4.5.1",
+		"sequelize": "6.33.0",
```
Comment (JamesP-KC) [inline packages/forms-web-app/package.json:60]: might not be needed. forms-web-app doesn't use it as far as i know

**Review by JamesP-KC — COMMENTED** [review]
Looks good, just a few npm dependencies that I think might have been moved around

File: packages/applications-service-api/src/controllers/subscriptions.js (hunk 2 above)
Comment (Yakitrak) [reply packages/applications-service-api/src/controllers/subscriptions.js:81]: Sorry I thought I removed this - will remove now.

File: package.json (hunk 3 above)
Comment (Yakitrak) [reply package.json:89]: All of this was added from me doing `npm update` and `npm audit` after updating node versions everywhere, not sure why its done but should I manually remove?

File: packages/forms-web-app/package.json (hunk 4 above)
Comment (Yakitrak) [reply packages/forms-web-app/package.json:60]: Was also added automatically but yes I cant see it on this package so not sure why it was added - will remove manually

File: packages/back-office-subscribers/package.json (hunk 1 above)
Comment (JamesP-KC) [reply packages/back-office-subscribers/package.json:15]: yeah i figured it was some automatic thing. Might be best to remove manually. Should only need `lodash.pick` out of those for this package

**Review by JamesP-KC — APPROVED** [review]

**Pull request merged**